upp/classes/components.py

```diff
@@ -11,21 +11,26 @@
 
 @dataclass
 class Component:
+    # (DONE) probably need equal_jets_from_samples here too
     region: Region
     sample: Sample
     flavour: Flavour
     global_cuts: Cuts
     dirname: Path
     num_jets: int
     num_jets_estimate: int
+    equal_jets_from_samples: bool
 
     def __post_init__(self):
         self.hist = Hist(self.dirname.parent.parent / "hists" / f"hist_{self.name}.h5")
 
     def setup_reader(self, batch_size, fname=None):
+        # (Done) equal_jets_from_samples here?
         if fname is None:
             fname = self.sample.path
-        self.reader = H5Reader(fname, batch_size)
+        self.reader = H5Reader(
+            fname, batch_size, equal_jets_from_samples=self.equal_jets_from_samples
```

Comment:
Idk if it's possible to find a variable name that uses less characters... maybe just `equal_samples`? In any case it would be good to document in a comment somewhere what this does

Reply:
I would probably use a default value that preserves the current behaviour